lib/lambda_ethereum_consensus/store/checkpoint_states_db.ex

```diff
@@ -41,18 +45,20 @@ defmodule LambdaEthereumConsensus.Store.CheckpointStates do
   """
   @spec get_checkpoint_state(Checkpoint.t()) :: {:ok, BeaconState.t()} | {:error, binary()}
   def get_checkpoint_state(checkpoint) do
-    case get(checkpoint) do
-      {:ok, state} ->
-        {:ok, state}
-
-      :not_found ->
-        with {:ok, state} <- compute_target_checkpoint_state(checkpoint.epoch, checkpoint.root) do
-          put(checkpoint, state)
-          {:ok, state}
-        end
+    case LRUCache.get(
+           @table,
+           checkpoint,
+           fn checkpoint -> compute_target_checkpoint_state(checkpoint.epoch, checkpoint.root) end
```

Comment:
In this line, we need to do something like this:
```elixir
with {:ok, value} <- compute_target_checkpoint_state(checkpoint.epoch, checkpoint.root) do
    value
end
```
Or to change the return type of `compute_target_checkpint_state`, otherwise we will have doubly indented results: 
```elixir
{:ok, {:ok, value}}
````

Reply:
Changed the whole code, as it's now an ets.